File: civitasRuby/pkg/civitasRuby-0.0.1/lib/civitas/sorpresa_pagar_cobrar.rb
#encoding:utf-8
=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

module Civitas

class SorpresaPagarCobrar < Sorpresa
  
  def initialize (valor, texto)
    super("*Pagar/Cobrar*", nil, valor, texto, nil)
  end
  
  def aplicarAJugador (actual, todos)
      if jugadorCorrecto(actual, todos)
        informe(actual, todos)
        todos[actual].modificarSaldo(@valor)
      end
    end
end

end

File: P5/civitasJava/src/civitas/SorpresaPagarCobrar.java

package civitas;

import java.util.ArrayList;

public class SorpresaPagarCobrar extends Sorpresa {
    
    SorpresaPagarCobrar(int valor, String texto) {
        super ("*Pagar/cobrar*");
        this.texto = texto;
        this.valor = valor;
    }
    
    /** Se modifica el saldo del jugador actual con el valor de la sorpresa. */
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe (actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
}
